crates/ink/ir/src/ir/chain_extension.rs

```diff
@@ -158,39 +221,100 @@ impl<'a> Iterator for ChainExtensionMethodInputs<'a> {
     }
 }
 
-/// The unique ID of an ink! chain extension method.
+/// The unique ID of an chain extension.
 ///
 /// # Note
 ///
-/// The ink! attribute `#[ink(extension = N: u32)]` for chain extension methods.
-///
-/// Has a `func_id` extension ID to identify the associated chain extension method.
-#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+/// The ink! attribute `#[ink::chain_extension(extension = N: u16)]` for chain extension.
+#[derive(Default, Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct ExtensionId {
-    index: u32,
+    index: u16,
 }
 
 impl ExtensionId {
-    /// Creates a new chain extension method ID from the given `u32`.
-    pub fn from_u32(index: u32) -> Self {
+    /// Creates a new chain extension ID from the given `u16`.
+    pub fn from_u16(index: u16) -> Self {
+        Self { index }
+    }
+
+    /// Returns the underlying raw `u16` index.
+    pub fn into_u16(self) -> u16 {
+        self.index
+    }
+}
+
+/// The unique ID of the method within the chain extension.
+///
+/// # Note
+///
+/// The ink! attribute `#[ink(function = N: u16)]` for chain extension methods.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct FunctionId {
+    index: u16,
+}
+
+impl FunctionId {
+    /// Creates a new chain extension function ID from the given `u16`.
+    pub fn from_u16(index: u16) -> Self {
         Self { index }
     }
 
+    /// Returns the underlying raw `u16` index.
+    pub fn into_u16(self) -> u16 {
+        self.index
+    }
+}
+
+/// The unique ID of a chain extension method across all chain extensions.
```

Comment:
The uniqueness of the id doesn't seem to be guaranteed anywhere, so we're just relying on the contract author to check for colliding ids? 

Tricky to do statically, though could possibly check for duplicates during the metadata pass.

Anyway it may not be important, since if by chance (unlikely?) there are colliding ids that will be caught very quickly.  when testing.